model/flow/header.go

```diff
@@ -104,6 +105,51 @@ func (h *Header) UnmarshalJSON(data []byte) error {
 	return err
 }
 
+// MarshalJSON makes sure the timestamp is encoded in UTC.
+func (h Header) MarshalCBOR() ([]byte, error) {
+
+	// NOTE: this is just a sanity check to make sure that we don't get
+	// different encodings if someone forgets to use UTC timestamps
+	if h.Timestamp.Location() != time.UTC {
+		h.Timestamp = h.Timestamp.UTC()
+	}
+
+	// we use an alias to avoid endless recursion; the alias will not have the
+	// marshal function and encode like a raw header
+	type Encodable Header
+	opts := cbor.CoreDetEncOptions() // CBOR deterministic options
+	// default: "2021-07-06 21:20:00 +0000 UTC" <- unwanted
+	// option : "2021-07-06 21:20:00.820603 +0000 UTC" <- wanted
+	opts.Time = cbor.TimeRFC3339Nano // option needed for wanted time format
```

Comment:
these seems like some sort of config that shouldn't be called on every MarshalCBOR call.

Reply:
@ramtinms I checked with @fxamacker and it's safe to pre-create the `CBOR` config and use them concurrently if need be. So I just pushed a change which does that :-)

Reply:
> @ramtinms I checked with @fxamacker and it's safe to pre-create the `CBOR` config and use them concurrently if need be. So I just pushed a change which does that :-)

Here's a bit more context in case others stumble across this.  😄

When`EncMode` is created, it makes an immutable copy of `EncOptions`.  By using its own internal copy of immutable options, `EncMode` avoids concurrency pitfalls.  So it's best to reuse a non-exported global `EncMode` that is created during startup.